ktor-client/ktor-client-plugins/ktor-client-auth/common/test/io/ktor/client/plugins/auth/AuthTest.kt

```diff
@@ -600,6 +601,38 @@ class AuthTest : ClientLoader() {
         }
     }
 
+    @Test
+    fun testRefreshAfterException() = clientTests {
+        var firstCall = true
+        config {
+            install(Auth) {
+                bearer {
+                    loadTokens { BearerTokens("first", "first") }
+
+                    refreshTokens {
+                        if (firstCall) {
+                            firstCall = false
+                            throw IOException("Refresh failed")
+                        }
+                        val token = client.get("$TEST_SERVER/auth/bearer/token/second?delay=500").bodyAsText()
+                        BearerTokens(token, token)
+                    }
+                }
+            }
+        }
+        test { client ->
+            firstCall = true
+            client.get("$TEST_SERVER/auth/bearer/first").bodyAsText()
+
+            assertFails {
```

Comment:
And also, maybe we want to check the message of an error to be sure that it is the expected error